docs/gcp_alert_policy_labels.md

```diff
@@ -0,0 +1,28 @@
+# GCP Alert Policy Labels
+> ## How Revere understands Google Cloud Monitoring Alert Policies
+
+Revere parses incoming Cloud Monitoring Alert incidents based on the "user labels" associated with the Alert Policy itself.
+
+This avoids using inconsistent naming schemes or the added complexity of attempting to parse alert conditions themselves.
+
+Label keys and values are subject to [Google's format and content restrictions](https://cloud.google.com/resource-manager/docs/creating-managing-labels#requirements).
+
+| Key | Meaning | Example Value |
+|:---:|:-------:|:-------:|
+| `revere-service-name` | "What is the 'short name' of the service?" | `buffer` |
+| `revere-service-environment` | "Where does this instance of the service operate?" | `prod` |
+| `revere-service-degradation` | "What is degraded when this alert policy triggers?" | `uptime` |
+
+The label values do not need to be unique: multiple alert policies can have the same labels to be understood the same way by Revere.
+
+### Failure Modes
+
+If Revere receives a Alert Policy incident notification and cannot find at least one of the above labels, it will attempt to parse the name of the Alert Policy to "fill in" whatever labels are missing.
```

Comment:
I'm totally on board with slack notifications. The reason I wrote this out is thinking of the following:

> When Revere is deployed, we put in the config file that any 'sam' alert in 'prod' should affect all of Terra.
> Months later, we add the status page to the process docs, and Revere handles the "alert users" part of the current manual process.
> More months later, someone adds a new SAM alert and forgets to add Revere labels.
> Yet more months later at 4am, the new alert fires, and while the engineer goes about fixing the problem they are entirely unaware that Revere never notified users or put up the banner (either because of not seeing the slack message, or because the slack token expired or something and we didn't know).

I wonder if maybe doing both is the sweet spot, so we can fix the root issue going forward during our normal working hours while still having one fewer failure mode for Revere when it counts. On the other hand, I'm always down to write less code.

Reply:
> More months later, someone adds a new SAM alert and forgets to add Revere labels.
Yet more months later at 4am, the new alert fires, and while the engineer goes about fixing the problem they are entirely unaware that Revere never notified users or put up the banner (either because of not seeing the slack message, or because the slack token expired or something and we didn't know).

I think the answer to this scenario is "better coupling between Revere and PD in Terraform". I.e., rather than asking devs to think about Revere at all, the standard Terraform module should configure sensible Revere labels by default for any alerting policy that forwards to PagerDuty.

If we support both names and labels, there's no incentive to follow the standard labeling scheme :) We're just asking for inconsistency.

Reply:
That's fair. Maybe the submodules in TF don't do any of this logic at all and they just all expose labels, and labels get made up above.